test/ExternalTest.ts

```diff
@@ -0,0 +1,161 @@
+                done();
+            });
+        };
+
+        const virtualAlexa = VirtualAlexa.Builder()
+            .handler(myFunction)
+            .sampleUtterancesFile("./test/resources/SampleUtterances.txt")
+            .intentSchemaFile("./test/resources/IntentSchema.json")
+            .create();
+
+        virtualAlexa.addressAPI().returnsFullAddress({
+            addressLine1: "address line 1",
+            addressLine2: "address line 2",
+            addressLine3: "address line 3",
+            city: "city",
+            countryCode: "country code",
+            districtOrCounty: "district",
+            postalCode: "postal",
+            stateOrRegion: "state",
+        });
+
+        virtualAlexa.launch().then();
+    });
+
+    it("Calls address API for country code", (done) => {
+        const myFunction = function(event: any, context: any) {
+            callAddressAPI(event.context.System.apiAccessToken, event.context.System.device.deviceId)
+                .then((response: any) => {
+                    assert.isUndefined(response.payload.addressLine1);
+                    assert.equal(response.payload.countryCode, "country code");
+                    assert.equal(response.statusCode, 200);
+                    done();
+                });
+        };
+
+        const virtualAlexa = VirtualAlexa.Builder()
+            .handler(myFunction)
+            .sampleUtterancesFile("./test/resources/SampleUtterances.txt")
+            .intentSchemaFile("./test/resources/IntentSchema.json")
+            .create();
+
+        virtualAlexa.addressAPI().returnsCountryAndPostalCode({
+            countryCode: "country code",
+            postalCode: "postal",
+        });
+
+        virtualAlexa.launch().then();
+    });
+
+    it("Calls with two different virtual alexas", (done) => {
+        // Make sure that only one mock for the address API is active at a time
+        let count = 0;
+        const myFunction = function(event: any, context: any) {
+            callAddressAPI(event.context.System.apiAccessToken, event.context.System.device.deviceId)
+                .then((response: any) => {
+                    count++;
+                    assert.isUndefined(response.payload.addressLine1);
+                    assert.equal(response.payload.countryCode, "country code " + count);
+                    assert.equal(response.statusCode, 200);
+                    if (count === 2) {
+                        done();
+                    }
+                });
+        };
+
+        let virtualAlexa = VirtualAlexa.Builder()
+            .handler(myFunction)
+            .sampleUtterancesFile("./test/resources/SampleUtterances.txt")
+            .intentSchemaFile("./test/resources/IntentSchema.json")
+            .create();
+
+        virtualAlexa.addressAPI().returnsCountryAndPostalCode({
+            countryCode: "country code 1",
+            postalCode: "postal",
+        });
+
+        virtualAlexa.launch().then();
```

Comment:
solution from before let you just do the .catch(error => done(error)) here and continue and then do the final done in the final launch avoiding any issue with timeouts if one of the launches fails